src/extra/AutoUIApp/Content.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { useLocation } from 'react-router';
+import { AutoUIModel, AutoUIBaseResource } from '../AutoUI/schemaOps';
+import { JSONSchema } from '../../components/Renderer/types';
+
+import { OpenApiJson } from './openApiJson';
+import {
+	getAllNaturalPropertiesKeys,
+	getSelectableOptions,
+	getSchemaFromLocation,
+	pine,
+} from './odata';
+import { useRequest } from '../../hooks/useRequest';
+import { AutoUI } from '../AutoUI';
+import { history, ObjectFromEntries } from '../AutoUI/utils';
+
+interface ContentProps {
+	openApiJson: OpenApiJson;
+}
+
+const generateModel = (
+	schema: JSONSchema,
+	resourceName: string,
+	naturalPropertiesKeys: string[],
+) => {
+	const [firstPropertyKey, ...otherPropertyKeys] = naturalPropertiesKeys;
+	return {
+		resource: resourceName,
+		schema,
+		permissions: {
+			read: [firstPropertyKey, ...otherPropertyKeys],
+			create: [],
+			update: [],
+			delete: false,
+		},
+		priorities: {
+			primary: [firstPropertyKey],
+			secondary: [...otherPropertyKeys],
+			tertiary: [],
+		},
+	} as AutoUIModel<AutoUIBaseResource<unknown>>;
+};
+
+export const Content = ({ openApiJson }: ContentProps) => {
+	const { pathname } = useLocation();
+	const resourceName = pathname.replace(/^\/([^\/]*).*$/, '$1');
+	const id = pathname.substr(pathname.lastIndexOf('/') + 1);
+	const endsWithValidId = !isNaN(parseInt(id, 10));
+	const schema = getSchemaFromLocation(openApiJson, resourceName);
+	const naturalPropertiesKeys = getAllNaturalPropertiesKeys(schema?.properties);
+	const correspondingPath = !endsWithValidId
+		? openApiJson.paths[`/${resourceName}`]
+		: openApiJson.paths[`/${resourceName}({id})`];
+
+	const [data] = useRequest(
+		async () =>
+			await pine.get({
+				resource: resourceName,
+				options: {
+					...(endsWithValidId && { id }),
+					$select: [...getSelectableOptions(correspondingPath, [])],
+				},
+			}),
+		[openApiJson, pathname],
+		{ polling: false },
+	);
+
+	const model = React.useMemo(() => {
+		if (!schema) {
+			return null;
+		}
+		return generateModel(schema, resourceName, naturalPropertiesKeys);
+	}, [schema, resourceName, naturalPropertiesKeys]);
+
+	if (!model) {
+		return <>Could not generate a model</>;
+	}
+
+	if (!(data as any[])?.length) {
+		return <>No data</>;
+	}
+
+	// TODO: since we don't have a way to render an object without a format yet, we will filter it.
```

Comment:
Can you clarify what you mean by "without a format" please? Doesn't the default entity lens for AutoUI successfully render an object?

Reply:
As you can see from the line below, I'm filtering an array of data, which is represented as a Collection. What I mean with my comment is that we still cannot render an object inside a cell of a collection